pantheon-sessions.php

```diff
@@ -342,12 +342,119 @@ public static function check_native_primary_keys() {
+	 *
+	 * @param int $start_position Site index to begin from.
+	 */
+	public function primary_key_finalize( $start_position ) {
+		global $wpdb;
+
+		if ( ! is_multisite() ) {
+			$this->safe_output( __( 'Single site detected. Beginning finalization.', 'wp-native-php-sessions' ), 'log' );
+
+			$this->add_single_index( $wpdb->prefix );
+
+			$this->safe_output( __( 'Operation complete.', 'wp-native-php-sessions' ), 'log' );
+
+			return;
+		}
+
+		$this->safe_output( __( 'Multisite installation detected. Processing Sites individually.', 'wp-native-php-sessions' ), 'log' );
+
+		$site_list = get_sites( [ 'number' => 500 ] );
+		$site_count = count( $site_list );
+
+		for ( $i = $start_position; $i < $site_count; $i++ ) {
+			// translators: $s is the array index of the current site.
+			$this->safe_output( __( 'Finalizing site %s. In the event of a timeout or error, resume execution starting from this point via "wp pantheon session add-index %s". To skip this blog if it does not need processing, run "wp pantheon session primary-key-finalize %s".', 'wp-native-php-sessions' ), 'log', [ $site_list[ $i ]->blog_id, $i, $i + 1 ] );
+
+			$this->primary_key_finalize_single( $wpdb->get_blog_prefix( $site_list[ $i ]->blog_id ), true );
+
+			// translators: %s is the current blog id.
+			$this->safe_output( __( 'Finalization of blog %s complete.', 'wp-native-php-sessions' ), 'log', [ $site_list[ $i ]->blog_id ] );
+		}
+	}
+
+	/**
+	 * Finalizes the creation of a primary key by deleting the old data.
+	 *
+	 * @param int $start_position Site index to begin from.
+	 */
+	public function primary_key_revert( $start_position ) {
+		global $wpdb;
+
+		if ( ! is_multisite() ) {
+			$this->safe_output( __( 'Single site detected. Beginning revert.', 'wp-native-php-sessions' ), 'log' );
+
+			$this->primary_key_revert_single( $wpdb->prefix );
+
+			$this->safe_output( __( 'Operation complete.', 'wp-native-php-sessions' ), 'log' );
+
+			return;
+		}
+
+		$this->safe_output( __( 'Multisite installation detected. Processing Sites individually.', 'wp-native-php-sessions' ), 'log' );
+
+		$site_list = get_sites( [ 'number' => 500 ] );
+		$site_count = count( $site_list );
+
+		for ( $i = $start_position; $i < $site_count; $i++ ) {
+			// translators: $s is the array index of the current site.
+			$this->safe_output( __( 'Processing site %s. In the event of a timeout or error, resume execution starting from this point via "wp pantheon session add-index %s". To skip this blog if it does not need processing, run "wp pantheon session primary-key-finalize %s".', 'wp-native-php-sessions' ), 'log', [ $site_list[ $i ]->blog_id, $i, $i + 1 ] );
+
+			$this->primary_key_revert_single( $wpdb->get_blog_prefix( $site_list[ $i ]->blog_id ), true );
+
+			// translators: %s is the current blog id.
+			$this->safe_output( __( 'Revert of blog %s complete.', 'wp-native-php-sessions' ), 'log', [ $site_list[ $i ]->blog_id ] );
```

Comment:
Less worried about "blog" vs "site" here (and other places where it's followed by an ID) because the value is _acually_ `blog_id`. However, `%s` could/should be replaced by `%d` if the value is a decimal. 🔢 

```suggestion
			$this->safe_output( __( 'Revert of blog %d complete.', 'wp-native-php-sessions' ), 'log', [ $site_list[ $i ]->blog_id ] );
```